packages/core/test/integration/query-interface/add-show-remove-index.test.ts

```diff
@@ -0,0 +1,304 @@
+          first_name: DataTypes.STRING,
+          last_name: DataTypes.STRING,
+        });
+
+      });
+
+      it('should add, show and remove a non-unique index', async () => {
+        await queryInterface.addIndex({ tableName: 'users', schema }, { name: 'unique_names', fields: ['username'], unique: false });
+
+        const indexes = await sequelize.queryInterface.showIndexes({ tableName: 'users', schema });
+        const indexCheck = indexes.find(index => index.name === 'unique_names');
+
+        expect(indexCheck).to.not.be.undefined;
+        assert(indexCheck);
+        if (['mariadb', 'mysql'].includes(dialect)) {
+          expect(indexCheck.schema).to.be.undefined;
+        } else {
+          expect(indexCheck.schema).to.equal(schema);
+        }
+
+        expect(indexCheck.tableName).to.equal('users');
+        expect(indexCheck.name).to.equal('unique_names');
+        expect(indexCheck.unique).to.equal(false);
+        expect(indexCheck.fields).to.have.length(1);
+        expect(indexCheck.fields[0].name).to.equal('username');
+
+        await queryInterface.removeIndex({ tableName: 'users', schema }, 'unique_names');
+        const indexesAfterRemoval = await sequelize.queryInterface.showIndexes({ tableName: 'users', schema });
+        const indexCheckAfterRemoval = indexesAfterRemoval.find(index => index.name === 'unique_names');
+
+        expect(indexCheckAfterRemoval).to.be.undefined;
+      });
+
+      it('should add, show and remove a unique index', async () => {
+        await queryInterface.addIndex({ tableName: 'users', schema }, { name: 'unique_names', fields: ['first_name', 'last_name'], unique: true });
+
+        const indexes = await sequelize.queryInterface.showIndexes({ tableName: 'users', schema });
+        const indexCheck = indexes.find(index => index.name === 'unique_names');
+
+        expect(indexCheck).to.not.be.undefined;
+        assert(indexCheck);
+        if (['mariadb', 'mysql'].includes(dialect)) {
+          expect(indexCheck.schema).to.be.undefined;
+        } else {
+          expect(indexCheck.schema).to.equal(schema);
+        }
+
+        expect(indexCheck.tableName).to.equal('users');
+        expect(indexCheck.name).to.equal('unique_names');
+        expect(indexCheck.unique).to.equal(true);
+        expect(indexCheck.fields).to.have.length(2);
+        expect(indexCheck.fields[0].name).to.equal('first_name');
+        expect(indexCheck.fields[1].name).to.equal('last_name');
+
+        await queryInterface.removeIndex({ tableName: 'users', schema }, 'unique_names');
+        const indexesAfterRemoval = await sequelize.queryInterface.showIndexes({ tableName: 'users', schema });
+        const indexCheckAfterRemoval = indexesAfterRemoval.find(index => index.name === 'unique_names');
+
+        expect(indexCheckAfterRemoval).to.be.undefined;
+      });
+    });
+  }
+
+  if (sequelize.dialect.supports.index.include) {
+    describe('Indexes with include', () => {
```

Comment:
could be merged with the first describe block, the "beforeEach" is identical

The tests can be conditional without needing a describe block

Reply:
Done